unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -1656,14 +1662,33 @@ loop = do
                 suffixifiedPPE
                   =<< makePrintNamesFromLabeled' (Patch.labeledDependencies patch)
               respond $ ListEdits patch ppe
-            PullRemoteBranchI mayRepo path syncMode verbosity -> unlessError do
+            PullRemoteBranchI mayRepo path syncMode pullMode verbosity -> unlessError do
+              let preprocess = case pullMode of
+                    Input.PullWithHistory -> pure
+                    Input.PullWithoutHistory -> pure . Branch.discardHistory
               ns <- maybe (writePathToRead <$> resolveConfiguredGitUrl Pull path) pure mayRepo
               lift $ unlessGitError do
-                b <- importRemoteBranch ns syncMode
-                let msg = Just $ PullAlreadyUpToDate ns path
+                remoteBranch <- importRemoteBranch ns syncMode preprocess
+                let unchangedMsg = PullAlreadyUpToDate ns path
                 let destAbs = resolveToAbsolute path
                 let printDiffPath = if Verbosity.isSilent verbosity then Nothing else Just path
-                lift $ mergeBranchAndPropagateDefaultPatch Branch.RegularMerge inputDescription msg b printDiffPath destAbs
+                lift $ case pullMode of
+                  Input.PullWithHistory -> do
+                    mergeBranchAndPropagateDefaultPatch
+                      Branch.RegularMerge
+                      inputDescription
+                      (Just unchangedMsg)
+                      remoteBranch
+                      printDiffPath
+                      destAbs
+                  Input.PullWithoutHistory -> do
```

Comment:
We don't propagate the patch here since we take the remote snapshot as-is.

Let me know if I'm missing something here.

Reply:
Currently, `pull.without-history remote local` conses a 'snapshot' of the remote's HEAD onto the local history.
So no, it won't ever **merge**, if you want merges, you're probably working on this project and you need to pull with history.
Think of pull.without-history like you're pulling a dependency of your project, if I upgrade from `random` version 1 to 1.5, I don't care about keeping the 'history' of `random`, I just want the latest version.

Reply:
Ok, I think I get it.